MDANSE_GUI/Src/MDANSE_GUI/Tabs/Models/TrajectoryModel.py

```diff
@@ -31,10 +33,43 @@
 )
 from qtpy.QtGui import QStandardItem, QStandardItemModel
 
+from MDANSE.Core.Platform import PLATFORM
 from MDANSE.MLogging import LOG
 from MDANSE.MolecularDynamics.Trajectory import Trajectory
 
 
+def store_recently_used_filename(
+    recent_file_path: str, max_num_of_files: int, loaded_file: str
+):
+    """Updating recent file list with successfully loaded file
+
+    Parameters
+    ----------
+    recent_file : str
+        JSON file path for storing recently loaded files
+    max_num_of_files : int
+        maximum number of files stored in recent file list
+    loaded_file : str
+        Path of the successfully loaded file
+    """
+    filepath = Path(recent_file_path)
+    # if the recent file exists and not empty
+    if filepath.is_file() and filepath.stat().st_size > 0:
+        with filepath.open(encoding="utf-8") as file:
+            recent_files = json.load(file)
+    else:
+        recent_files = []
```

Comment:
May be better duck typed.


```suggestion
    if filepath.is_file():
        try:
            with filepath.open(encoding="utf-8") as file:
                recent_files = json.load(file)
         except json.decoder.JSONDecodeError as err:
             log.error("Invalid recent files json at %s.", filepath) from err
             recent_files = []
    else:
        recent_files = []
```